File: DeepSpace-Java/src/test/TestP1.java
package deepspace;

//
// ─── PRUEBAS DE LAS CLASES ──────────────────────────────────────────────────────
//

public class TestP1 {
    public static void main(String[] args) {
    //
    // ───────────────────────────────────────────────────────────────────── LOOT ─────
    //

        Loot botin = new Loot(1,2,3,4,5);

        System.out.print("Prueba de interfaces de Loot:\n");
        System.out.printf( "-> Supplies: " + botin.getNSupplies() +
                         "\n-> Weapons: "  + botin.getNWeapons()  +
                         "\n-> Shields: "  + botin.getNShields()  +
                         "\n-> Hangars: "  + botin.getNHangars()  +
                         "\n-> Medals: "   + botin.getNMedals()   + "\n\n" );

    //
    // ─────────────────────────────────────────────────────────────────── WEAPON ─────
    //

        Weapon arma = new Weapon("Rayo", WeaponType.LASER, 1);

        System.out.print("Prueba de métodos de Weapon\n");
        System.out.printf("-> Tipo: "     + arma.getType() +
                        "\n-> Usos: "     + arma.getUses() +
                        "\n-> Potencia: " + arma.power()   + "\n");

        arma.useIt();
        System.out.printf("Tras useIt(), tiene estos usos: " + arma.getUses() + "\n\n");


    //
    // ──────────────────────────────────────────────────────────── SHIELDBOOSTER ─────
    //

        ShieldBooster escudo = new ShieldBooster("Ligero", 3.5f, 0);

        System.out.print("Prueba de métodos de ShieldBooster\n");
        System.out.printf("-> Boosteo: " + escudo.getBoost() +
                        "\n-> Usos: "    + escudo.getUses());

        escudo.useIt();
        System.out.printf("Tras useIt(), tiene estos usos: " + escudo.getUses() + "\n\n");

    //
    // ────────────────────────────────────────────────────────── SUPPLIESPACKAGE ─────
    //

        SuppliesPackage suministros = new SuppliesPackage(3.0f, 50f, 3.5f);

        System.out.print("Prueba de métodos de SuppliesPackage\n");
        System.out.printf("-> Ammo: "     + suministros.getAmmoPower() +
                        "\n-> Fuel: "     + suministros.getFuelUnits() +
                        "\n-> Potencia: " + suministros.getShieldPower() + "\n\n");

    //
    // ───────────────────────────────────────────────────────────────────── DICE ─────
    //

        Dice dado = new Dice();
        int jugadores = 5;
        float velocidad = 0.5f;

        System.out.print("Prueba de métodos de Dice\n");
        System.out.print("Jugadores: " + jugadores + "\n");
        System.out.print("Velocidad: " + velocidad + "\n");

        System.out.print("Haremos 100 iteraciones para ver la distribución de números aleatorios\n");

        int[] cuentaArmas        = {0, 0, 0};
        int[] cuentaHangares     = {0, 0};
        int[] cuentaEscudos      = {0, 0};
        int[] cuentaQuienDispara = {0, 0};
        int[] haMovido           = {0, 0};
        int[] cuentaQuienEmpieza = new int [jugadores];

        for (int i = 0; i < jugadores; i++) {
            cuentaQuienEmpieza[i] = 0;
        }

        for (int i=0; i < 100; i++) {
            cuentaHangares[dado.initWithNHangars()]++;
            cuentaEscudos[dado.initWithNShields()]++;
            cuentaArmas[dado.initWithNWeapons() - 1]++;
            cuentaQuienEmpieza[dado.whoStarts(jugadores)]++;

            if (dado.spaceStationMoves(velocidad))
                haMovido[1]++;
            else
                haMovido[0]++;

            if (dado.firstShot() == GameCharacter.ENEMYSTARSHIP)
                cuentaQuienDispara[0]++;
            else
                cuentaQuienDispara[1]++;
        }

        System.out.print("Hangares: 0 -> " + cuentaHangares[0] + ", 1 -> " + cuentaHangares[1] + "\n");
        System.out.print("Escudos: 0 -> " + cuentaEscudos[0] + ", 1 -> " + cuentaEscudos[1] + "\n");
        System.out.print("Armas: 1 -> " + cuentaArmas[0] + ", 2 -> " + cuentaArmas[1] + ", 3 -> " + cuentaArmas[2] + "\n");
        System.out.print("Quién dispara: EnemyStarship -> " + cuentaQuienDispara[0] + ", SpaceStation -> " + cuentaQuienDispara[1] + "\n");
        System.out.print("Ha conseguido mover: " + haMovido[1] + ". No lo ha conseguido: " + haMovido[0] + "\n");
        System.out.print("Qué jugador empieza: \n");

        for (int i = 0; i < jugadores; i++) {
            System.out.print("Jugador " + (i+1) +": " + cuentaQuienEmpieza[i] + "\n");
        }

    }
}

File: DeepSpace-Ruby/test/test_p1.rb
# encoding:utf-8

require '../lib/CombatResult.rb'
require '../lib/Dice.rb'
require '../lib/GameCharacter.rb'
require '../lib/Loot.rb'
require '../lib/ShieldBooster.rb'
require '../lib/ShotResult.rb'
require '../lib/SuppliesPackage.rb'
require '../lib/Weapon.rb'
require '../lib/WeaponType.rb'

module Deepspace
    class TestP1
        def initialize

        end

        def self.main

            botin = Loot.new(1,2,3,4,5)
            puts "Prueba de interfaces de Loot:"
            puts "-> Supplies: #{botin.nSupplies}\n-> NWeapons: #{botin.nWeapons}\n-> NShields: #{botin.nShields}\n-> Hangars: #{botin.nHangars}\n-> Medals: #{botin.nMedals}\n\n"

            arma = Weapon.new("Rayo", WeaponType::LASER, 1)
            puts "Prueba de métodos de Weapon:"
            puts "-> Tipo: #{arma.type}\n-> Usos: #{arma.uses}\n->Potencia: #{arma.power}"
            arma.useIt
            puts "Tras useIt tiene estos usos: #{arma.uses}\n\n"

            escudo = ShieldBooster.new("Ligero", 3.5, 0)
            puts "Prueba de métodos de ShieldBooster:"
            puts "-> Boost: #{escudo.boost}\n-> Usos: #{escudo.uses}\n\n"

            suministros = SuppliesPackage.new(3.0,50,3.5)
            puts "Prueba de métodos de SuppliesPackage"
            puts "-> Ammo: #{suministros.ammoPower}\n-> Fuel: #{suministros.fuelUnits}\n-> Potencia: #{suministros.shieldPower}\n\n"

            dado = Dice.new()
            jugadores = 5
            velocidad = 0.5

            puts "Prueba de métodos de Dice"
            puts "Jugadores: #{jugadores}"
            puts "Hacemos 100 iteraciones para ver la generación de números aleatorios:"

            cuenta_armas         = [0, 0, 0]
            cuenta_hangares      = [0, 0]
            cuenta_escudos       = [0, 0]
            cuenta_quien_dispara = [0, 0]
            ha_movido            = [0, 0]
            cuenta_quien_empieza = Array.new(jugadores, 0)

            for i in 1..100
                cuenta_hangares[dado.initWithNHangars]   += 1
                cuenta_escudos[dado.initWithNShields]    += 1
                cuenta_armas[dado.initWithNWeapons - 1]  += 1
                cuenta_quien_empieza[dado.whoStarts(jugadores) - 1] += 1

                if dado.spaceStationMoves(velocidad)
                    ha_movido[1] += 1
                else
                    ha_movido[0] += 1
                end

                if dado.firstShot() == GameCharacter::ENEMYSTARSHIP
                    cuenta_quien_dispara[0] += 1
                else
                    cuenta_quien_dispara[1] += 1
                end
            end

            puts "Hangares: 0 -> #{cuenta_hangares[0]}, 1 -> #{cuenta_hangares[1]}"
            puts "Escudos: 0 -> #{cuenta_escudos[0]}, 1 -> #{cuenta_escudos[1]}"
            puts "Armas: 1 -> #{cuenta_armas[0]}, 2 -> #{cuenta_armas[1]}, 3 -> #{cuenta_armas[2]}"
            puts "Quién dispara: ENEMYSTARSHIP #{cuenta_quien_dispara[0]}, SPACESTATION #{cuenta_quien_dispara[1]}"
            puts "Ha conseguido mover: #{ha_movido[1]}. No lo ha conseguido: #{ha_movido[0]}"
            puts "Qué jugador empieza:"

            for i in 0...jugadores
                puts "#{i + 1}: #{cuenta_quien_empieza[i]} veces"
            end

        end
   end

   TestP1.main
end
